libs/wire-api/test/unit/Test/Wire/API/Golden/Runner.hs

```diff
@@ -57,6 +64,44 @@ testObject obj path = do
 
   pure exists
 
+protoTestObjects ::
+  forall m a.
+  (Typeable a, ToProto a, FromProto a, Eq a, Show a, Show m, Eq m, Message m) =>
+  [(a, FilePath)] ->
+  IO ()
+protoTestObjects objs = do
+  allFilesExist <- and <$> traverse (uncurry (protoTestObject @m)) objs
+  assertBool "Some golden protobuf files do not exist" allFilesExist
+
+protoTestObject ::
+  forall m a.
+  (Typeable a, ToProto a, FromProto a, Eq a, Show a, Show m, Eq m, Message m) =>
+  a ->
+  FilePath ->
+  IO Bool
+protoTestObject obj path = do
+  let actual = toProto obj
+  msg <- assertRight (decodeMessage @m (LBS.toStrict actual))
+  let pretty = render (pprintMessage msg)
```

Comment:
I didn't know there was pretty print for protobuf, this is sweet :smile:

Reply:
Yes, I was really happy to find it. Otherwise, having golden files with binary content would make it quite hard to handle changes in the protocol.